Warnings in code

There are some new warnings in the code. gcc must have added these recently, because I've been seeing these warnings appear in other projects too:

```
test_pioc_fill.c: In function ‘test_fill’:
test_pioc_fill.c:550:66: warning: ‘%s’ directive output may be truncated writing up to 256 bytes into a region of size between 216 and 226 [-Wformat-truncation=]
  550 |             snprintf(filename, PIO_MAX_NAME, "%s_default_fill_%d_%s.nc", TEST_NAME, default_fill, iotype_name);
      |                                                                  ^~                               ~~~~~~~~~~~
test_pioc_fill.c:550:13: note: ‘snprintf’ output between 34 and 300 bytes into a destination of size 256
  550 |             snprintf(filename, PIO_MAX_NAME, "%s_default_fill_%d_%s.nc", TEST_NAME, default_fill, iotype_name);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/bin/bash ../../libtool  --tag=CC   --mode=link mpicc  -g -Wall  -L/usr/local/pnetcdf-1.12.1/lib -L/usr/local/netcdf-c-4.8.0-development_hdf5-1.10.6_szip_mpich/lib -L/usr/local/netcdf-fortran-4.5.4-development_szip_mpich/lib -o test_async_4proc test_async_4proc.o test_common.o ../../src/clib/libpioc.la -lpnetcdf -lnetcdf 
test_pioc_fill.c: In function ‘test_fill_mode’:
test_pioc_fill.c:656:106: warning: ‘%s’ directive output may be truncated writing up to 256 bytes into a region of size between 166 and 196 [-Wformat-truncation=]
  656 |                     snprintf(filename, PIO_MAX_NAME, "%s_fill_mode_async_%d_default_fill_%d_extra_var_%d_%s.nc",
      |                                                                                                          ^~
  657 |                              TEST_NAME, async, default_fill, extra_var, iotype_name);
      |                                                                         ~~~~~~~~~~~                       
test_pioc_fill.c:656:21: note: ‘snprintf’ output between 64 and 350 bytes into a destination of size 256
  656 |                     snprintf(filename, PIO_MAX_NAME, "%s_fill_mode_async_%d_default_fill_%d_extra_var_%d_%s.nc",
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  657 |                              TEST_NAME, async, default_fill, extra_var, iotype_name);
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mv -f .deps/test_iosystem3.Tpo .deps/test_iosystem3.Po
test_pioc.c: In function ‘test_names’:
test_pioc.c:850:31: warning: ‘%s’ directive writing up to 256 bytes into a region of size 247 [-Wformat-overflow=]
  850 |         sprintf(filename, "%s_%s_names.nc", TEST_NAME, iotype_name);
      |                               ^~                       ~~~~~~~~~~~
test_pioc.c:850:9: note: ‘sprintf’ output between 20 and 276 bytes into a destination of size 257
  850 |         sprintf(filename, "%s_%s_names.nc", TEST_NAME, iotype_name);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mv -f .deps/test_pioc_putget.Tpo .deps/test_pioc_putget.Po
test_pioc.c: In function ‘test_files’:
test_pioc.c:966:31: warning: ‘%s’ directive writing up to 256 bytes into a region of size 247 [-Wformat-overflow=]
  966 |         sprintf(filename, "%s_%s.nc", TEST_NAME, iotype_name);
      |                               ^~                 ~~~~~~~~~~~
test_pioc.c:966:9: note: ‘sprintf’ output between 14 and 270 bytes into a destination of size 257
  966 |         sprintf(filename, "%s_%s.nc", TEST_NAME, iotype_name);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test_pioc.c: In function ‘test_empty_files’:
test_pioc.c:1050:37: warning: ‘%s’ directive writing up to 256 bytes into a region of size 241 [-Wformat-overflow=]
 1050 |         sprintf(filename, "%s_empty_%s.nc", TEST_NAME, iotype_name);
      |                                     ^~                 ~~~~~~~~~~~
test_pioc.c:1050:9: note: ‘sprintf’ output between 20 and 276 bytes into a destination of size 257
 1050 |         sprintf(filename, "%s_empty_%s.nc", TEST_NAME, iotype_name);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test_pioc.c: In function ‘test_find_var_fillvalue’:
test_pioc.c:1249:50: warning: ‘%s’ directive writing up to 256 bytes into a region of size 228 [-Wformat-overflow=]
 1249 |         sprintf(filename, "%s_find_var_fillvalue_%s.nc", TEST_NAME, iotype_name);
      |                                                  ^~                 ~~~~~~~~~~~
test_pioc.c:1249:9: note: ‘sprintf’ output between 33 and 289 bytes into a destination of size 257
 1249 |         sprintf(filename, "%s_find_var_fillvalue_%s.nc", TEST_NAME, iotype_name);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test_pioc.c: In function ‘test_deletefile’:
test_pioc.c:1392:41: warning: ‘%s’ directive writing up to 256 bytes into a region of size 237 [-Wformat-overflow=]
 1392 |         sprintf(filename, "delete_me_%s_%s.nc", TEST_NAME, iotype_name);
      |                                         ^~                 ~~~~~~~~~~~
test_pioc.c:1392:9: note: ‘sprintf’ output between 24 and 280 bytes into a destination of size 257
 1392 |         sprintf(filename, "delete_me_%s_%s.nc", TEST_NAME, iotype_name);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test_pioc.c: In function ‘test_nc4’:
test_pioc.c:1465:31: warning: ‘%s’ directive writing up to 256 bytes into a region of size 247 [-Wformat-overflow=]
 1465 |         sprintf(filename, "%s_%s.nc", TEST_NAME, iotype_name);
      |                               ^~                 ~~~~~~~~~~~
test_pioc.c:1465:9: note: ‘sprintf’ output between 14 and 270 bytes into a destination of size 257
 1465 |         sprintf(filename, "%s_%s.nc", TEST_NAME, iotype_name);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test_pioc.c: In function ‘test_scalar’:
test_pioc.c:1802:31: warning: ‘%s’ directive writing up to 256 bytes into a region of size 247 [-Wformat-overflow=]
 1802 |         sprintf(filename, "%s_%s_scalar_async_%d.nc", TEST_NAME, iotype_name, async);
      |                               ^~                                 ~~~~~~~~~~~
test_pioc.c:1802:9: note: ‘sprintf’ output between 29 and 295 bytes into a destination of size 257
 1802 |         sprintf(filename, "%s_%s_scalar_async_%d.nc", TEST_NAME, iotype_name, async);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

--- a/tests/cunit/test_async_3proc.c
+++ b/tests/cunit/test_async_3proc.c
@@ -71,7 +71,7 @@ int main(int argc, char **argv)
             {
                 for (int flv = 0; flv < num_flavors; flv++)
                 {
-                    char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+                    char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
                     int my_comp_idx = 0; /* Index in iosysid array. */
 
                     for (int sample = 0; sample < NUM_SAMPLES; sample++)

--- a/tests/cunit/test_async_4proc.c
+++ b/tests/cunit/test_async_4proc.c
@@ -69,7 +69,7 @@ int main(int argc, char **argv)
             {
                 for (int flv = 0; flv < num_flavors; flv++)
                 {
-                    char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+                    char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
                     int my_comp_idx = 0; /* Index in iosysid array. */
 
                     for (int sample = 0; sample < NUM_SAMPLES; sample++)

--- a/tests/cunit/test_async_simple.c
+++ b/tests/cunit/test_async_simple.c
@@ -90,7 +90,7 @@ int main(int argc, char **argv)
 
                 for (int sample = 0; sample < NUM_SAMPLES; sample++)
                 {
-                    char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+                    char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
                     char iotype_name[PIO_MAX_NAME + 1];
 
                     /* Create a filename. */

--- a/tests/cunit/test_pioc.c
+++ b/tests/cunit/test_pioc.c
@@ -838,7 +838,7 @@ int test_names(int iosysid, int num_flavors, int *flavor, int my_rank,
     {
         int ncid;
         int varid;
-        char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+        char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
         char iotype_name[PIO_MAX_NAME + 1];
         int dimids[NDIM];        /* The dimension IDs. */
         int att_val = ATT_VAL;
@@ -942,7 +942,7 @@ int test_files(int iosysid, int num_flavors, int *flavor, int my_rank)
      * available ways. */
     for (int fmt = 0; fmt < num_flavors; fmt++)
     {
-        char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+        char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
         char iotype_name[PIO_MAX_NAME + 1];
 
         /* Overwrite existing test file. */
@@ -1041,7 +1041,7 @@ int test_empty_files(int iosysid, int num_flavors, int *flavor, int my_rank)
      * available ways. */
     for (int fmt = 0; fmt < num_flavors; fmt++)
     {
-        char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+        char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
         char iotype_name[PIO_MAX_NAME + 1];
 
         /* Create a filename. */
@@ -1218,7 +1218,7 @@ int test_find_var_fillvalue(int iosysid, int num_flavors, int *flavor,
      * available ways. */
     for (int fmt = 0; fmt < num_flavors; fmt++)
     {
-        char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+        char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
         char iotype_name[PIO_MAX_NAME + 1];
         int num_types = NUM_CLASSIC_TYPES;
 
@@ -1370,7 +1370,7 @@ int test_deletefile(int iosysid, int num_flavors, int *flavor, int my_rank)
      * available ways. */
     for (int fmt = 0; fmt < num_flavors; fmt++)
     {
-        char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+        char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
         char iotype_name[PIO_MAX_NAME + 1];
         int old_method;
 
@@ -1456,7 +1456,7 @@ int test_nc4(int iosysid, int num_flavors, int *flavor, int my_rank)
      * available ways. */
     for (int fmt = 0; fmt < num_flavors; fmt++)
     {
-        char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+        char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
         char iotype_name[PIO_MAX_NAME + 1];
 
         /* Create a filename. */
@@ -1793,7 +1793,7 @@ int test_scalar(int iosysid, int num_flavors, int *flavor, int my_rank, int asyn
      * available ways. */
     for (int fmt = 0; fmt < num_flavors; fmt++)
     {
-        char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+        char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
         char iotype_name[PIO_MAX_NAME + 1];
 
         /* Create a filename. */

--- a/tests/cunit/test_pioc_fill.c
+++ b/tests/cunit/test_pioc_fill.c
@@ -538,7 +538,7 @@ int test_fill(int iosysid, int num_flavors, int *flavor, int my_rank,
          * available ways. */
         for (int fmt = 0; fmt < num_flavors; fmt++)
         {
-            char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+            char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
             char iotype_name[PIO_MAX_NAME + 1];
             int ncid;
             int varid[NUM_NETCDF_TYPES];
@@ -547,7 +547,8 @@ int test_fill(int iosysid, int num_flavors, int *flavor, int my_rank,
             /* Create a filename. */
             if ((ret = get_iotype_name(flavor[fmt], iotype_name)))
                 ERR(ret);
-            snprintf(filename, PIO_MAX_NAME, "%s_default_fill_%d_%s.nc", TEST_NAME, default_fill, iotype_name);
+            snprintf(filename, PIO_MAX_NAME * 2, "%s_default_fill_%d_%s.nc", TEST_NAME,
+		     default_fill, iotype_name);
 
             /* Create test file with dims and vars defined. */
             if ((ret = create_putget_file(iosysid, flavor[fmt], dim_len, varid, filename,
@@ -635,7 +636,7 @@ int test_fill_mode(int iosysid, int num_flavors, int *flavor, int my_rank,
             {
                 for (int t = 0; t < NUM_TYPES_TO_TEST; t++)
                 {
-                    char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+                    char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
                     char iotype_name[PIO_MAX_NAME + 1];
                     int ncid;
                     int dimid;
@@ -653,7 +654,7 @@ int test_fill_mode(int iosysid, int num_flavors, int *flavor, int my_rank,
                     /* Create a filename. */
                     if ((ret = get_iotype_name(flavor[fmt], iotype_name)))
                         ERR(ret);
-                    snprintf(filename, PIO_MAX_NAME, "%s_fill_mode_async_%d_default_fill_%d_extra_var_%d_%s.nc",
+                    snprintf(filename, PIO_MAX_NAME * 2, "%s_fill_mode_async_%d_default_fill_%d_extra_var_%d_%s.nc",
                              TEST_NAME, async, default_fill, extra_var, iotype_name);
 
                     /* Create the test file. */

--- a/tests/cunit/test_pioc_putget.c
+++ b/tests/cunit/test_pioc_putget.c
@@ -321,7 +321,7 @@ int test_atts_byte(int iosysid, int num_flavors, int *flavor, int my_rank,
     for (int fmt = 0; fmt < num_flavors; fmt++)
     {
         char iotype_name[PIO_MAX_NAME + 1];
-        char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+        char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
         int ncid;
         int ret;    /* Return code. */
 
@@ -433,7 +433,7 @@ int test_atts_int64(int iosysid, int num_flavors, int *flavor, int my_rank,
     for (int fmt = 0; fmt < num_flavors; fmt++)
     {
         char iotype_name[PIO_MAX_NAME + 1];
-        char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+        char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
         int ncid;
         int ret;    /* Return code. */
 
@@ -1917,7 +1917,7 @@ int test_putget(int iosysid, int num_flavors, int *flavor, int my_rank,
              * available ways. */
             for (int fmt = 0; fmt < num_flavors; fmt++)
             {
-                char filename[PIO_MAX_NAME + 1]; /* Test filename. */
+                char filename[PIO_MAX_NAME * 2 + 1]; /* Test filename. */
                 char iotype_name[PIO_MAX_NAME + 1];
                 int ncid;
                 int varid[NUM_NETCDF4_TYPES + 1];
@@ -1926,7 +1926,7 @@ int test_putget(int iosysid, int num_flavors, int *flavor, int my_rank,
                 /* Create a filename. */
                 if ((ret = get_iotype_name(flavor[fmt], iotype_name)))
                     return ret;
-                snprintf(filename, PIO_MAX_NAME, "%s_putget_access_%d_unlim_%d_%s.nc", TEST_NAME,
+                snprintf(filename, PIO_MAX_NAME * 2, "%s_putget_access_%d_unlim_%d_%s.nc", TEST_NAME,
                          access, unlim, iotype_name);
 
                 /* Create test file with dims and vars defined. */

